File: Lab_04/C++/MyClass.cpp
#include "MyClass.h"
#include <windows.h>


MyClass::MyClass() {
	init();
}

MyClass::MyClass(vector<char> str) {
	this->str = str;
	length(str);
}

MyClass::MyClass(MyClass& copy) {
	this->str = copy.str;
	this->len = copy.str.size();
}

vector<char> MyClass::getStr() {
	return this->str;
}

void MyClass::length(vector<char> str) {
	this->len = str.size();
}

void MyClass::init() {
	this->str.push_back('\0');
	len = 0;
}

MyClass MyClass::operator +(MyClass first) {
	MyClass result;
	result.str = this->str;

	try {
		if (first.len == 0) {
			throw "Entered string is empty";
		}
	}

	catch (const char* ch) {
		MessageBoxA(GetActiveWindow(), ch, "Error", MB_ICONERROR);
		//exit(1);
	}

	for (int i = 0; i < first.len; i++) {
		result.str.push_back(first.str[i]);
	}
	result.len = first.len + this->len;
	return result;
}

MyClass MyClass::operator /(int num) {
	MyClass result;


	vector<char> temp_str;
	for (int i = 0; i < this->len; i++) {
		if (i % num == 0) {
			temp_str.push_back(this->str[i]);
		}
	}
	this->str = temp_str;
	length(this->str);
	result.str = this->str;
	result.len = this->len;
	return result;
}

File: Lab_04/C#/MyClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class MyClass
{
    private int len;
    private string str;

    private void init()
    {
        str = "";
        len = 0;
    }

    public MyClass()
    {
        init();
    }

    public MyClass(string str)
    {
        this.str = str;
        length(str);
    }

    public MyClass(MyClass copy)
    {
        this.str = copy.str;
        this.len = copy.str.Length;
    }

    public void length(string str)
    {
        this.len = str.Length;
    }

    public string getStr()
    {
        return this.str;
    }

    public static MyClass operator +(MyClass first, MyClass second)
    {
        if (first.str.Length == 0 || second.str.Length == 0)
        {
            throw new ArgumentNullException("Entered string is empty");
        }

        MyClass result = new MyClass();
        result.str = first.str + second.str;
        return result;
    }

    public static MyClass operator /(MyClass str_val, int num)
    {
        if (str_val.str.Length == 0)
        {
            throw new ArgumentNullException("Entered string is empty");
        }


        string temp_str = "";
        for (int i = 0; i < str_val.len; i++)
        {
            if (i % num == 0)
            {
                temp_str += str_val.str[i];
            }
        }
        str_val.str = temp_str;
        return str_val;
    }
}
